integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java

```diff
@@ -109,25 +109,47 @@ public static void main(String[] args) {
           + "Proceed with local configuration for FUSE: {}", e.toString());
     }
     conf = fsContext.getClusterConf();
-    final AlluxioFuseOptions opts = parseOptions(args, conf);
+    final FuseMountInfo opts = parseOptions(args, conf);
     if (opts == null) {
       System.exit(1);
     }
+    try {
+      launchFuse(fsContext, opts, true);
+    } catch (IOException e) {
+      LOG.error(e.getMessage());
+    }
+  }
+
+  /**
+   * Launches Fuse application.
+   *
+   * @param fsContext file system context for Fuse client to communicate to servers
+   * @param opts the fuse mount options
+   * @param blocking whether the Fuse application is blocking or not
+   * @return the Fuse application handler for future unmount operation
+   */
+  public static FuseUnmountable launchFuse(FileSystemContext fsContext,
+      FuseMountInfo opts, boolean blocking) throws IOException {
+    if (opts == null) {
+      throw new IOException("The given fuse options cannot be null");
+    }
+    AlluxioConfiguration conf = fsContext.getClusterConf();
     try (final FileSystem fs = FileSystem.Factory.create(fsContext)) {
       final List<String> fuseOpts = opts.getFuseOpts();
       if (conf.getBoolean(PropertyKey.FUSE_JNIFUSE_ENABLED)) {
         final AlluxioJniFuseFileSystem fuseFs = new AlluxioJniFuseFileSystem(fs, opts, conf);
         try {
           LOG.info("Mounting AlluxioJniFuseFileSystem: mount point=\"{}\", OPTIONS=\"{}\"",
               opts.getMountPoint(), fuseOpts.toArray(new String[0]));
-          fuseFs.mount(true, opts.isDebug(), fuseOpts.toArray(new String[0]));
+          fuseFs.mount(blocking, opts.isDebug(), fuseOpts.toArray(new String[0]));
         } catch (FuseException e) {
-          LOG.error("Failed to mount {}", opts.getMountPoint(), e);
           // only try to umount file system when exception occurred.
           // jni-fuse registers JVM shutdown hook to ensure fs.umount()
           // will be executed when this process is exiting.
           fuseFs.umount();
+          throw new IOException(String.format("Failed to mount %s", opts.getMountPoint()), e);
```

Comment:
Will this print out the local machine mount point as well as the alluxio directory?

Reply:
Usually for fuse mount, the identifier is the local machine mount point which is unique. Several mount point can point to the same alluxio directory. 
but i can add the alluxio directory to this error message.